### ISSUE TITLE
Storage changes mismatch @mainnet's block 4188122

### ISSUE THREAD

**Issue by roman-khimov** [nspcc-dev/neo-go#978]
file BlockStorage_4200000/dump-block-4189000.json: block 4188122, changes number mismatch: 0 vs 2

I'll add more info.

**Comment by roman-khimov** [comment]
Theirs:
```json
{
   "id" : 1,
   "result" : {
      "executions" : [
         {
            "notifications" : [
               {
                  "contract" : "0xe48dde213ee6e51cbc0a888339b335dc6122d401",
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "value" : "4f6e486f6c6443726561746564",
                           "type" : "ByteArray"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "3a6b34ac7fe77871e5a3a92349959dc6c7bd88fd"
                        },
                        {
                           "value" : "e72d286979ee6cb1b7e65dfddfb2e384100b8d148e7758de42e4168b71792c60",
                           "type" : "ByteArray"
                        },
                        {
                           "value" : [
                              {
                                 "type" : "ByteArray",
                                 "value" : "406da70000000000"
                              }
                           ],
                           "type" : "Array"
                        },
                        {
                           "type" : "Array",
                           "value" : [
                              {
                                 "value" : "90d801f1c61579feb2d3fe7e681985e585df6570b0a6f15c36c50c7fa7e18b2d",
                                 "type" : "ByteArray"
                              }
                           ]
                        },
                        {
                           "value" : [
                              {
                                 "type" : "ByteArray",
                                 "value" : ""
                              }
                           ],
                           "type" : "Array"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "3a6b34ac7fe77871e5a3a92349959dc6c7bd88fd0200000000000000"
                        }
                     ]
                  }
               }
            ],
            "vmstate" : "HALT",
            "contract" : "0x4b2258bc140e22b86c3eb83666ce70b5a52cb037",
            "gas_consumed" : "2.357",
            "stack" : [
               {
                  "type" : "Integer",
                  "value" : "1"
               }
            ],
            "trigger" : "Application"
         }
      ],
      "txid" : "0xdfa793e2cb400bc41819bbd2e33ca9d39ab1442a60047b2de5fea940ad27b1bf"
   },
   "jsonrpc" : "2.0"
}
```

Ours:
```json
{
   "result" : {
      "txid" : "0xdfa793e2cb400bc41819bbd2e33ca9d39ab1442a60047b2de5fea940ad27b1bf",
      "executions" : [
         {
            "vmstate" : "FAULT",
            "stack" : [],
            "contract" : "0x4b2258bc140e22b86c3eb83666ce70b5a52cb037",
            "gas_consumed" : "1.086",
            "trigger" : "Application",
            "notifications" : []
         }
      ]
   },
   "jsonrpc" : "2.0",
   "id" : 1
}
```

**Comment by fyrchik** [comment]
Closed via #979 .

**Comment by roman-khimov** [comment]
Seems like there is still something wrong:
file BlockStorage_4200000/dump-block-4189000.json: block 4188122, changes number mismatch: 0 vs 2

Version:
```
$ ./bin/neo-go --version
neo-go version 0.74.1-pre-237-gff0241ed
```

Log:
```
2020-05-23T10:40:21.852+0300    WARN    contract invocation failed      {"tx": "dfa793e2cb400bc41819bbd2e33ca9d39ab1442a60047b2de5fea940ad27b1bf", "block": 4188122, "error": "error encountered at instruction 10417 (THROW): THROW"}
```

**Comment by roman-khimov** [comment]
New app log:
```
{
   "result" : {
      "txid" : "0xdfa793e2cb400bc41819bbd2e33ca9d39ab1442a60047b2de5fea940ad27b1bf",
      "executions" : [
         {
            "contract" : "0x4b2258bc140e22b86c3eb83666ce70b5a52cb037",
            "gas_consumed" : "1.1",
            "vmstate" : "FAULT",
            "trigger" : "Application",
            "stack" : [
               {
                  "type" : "ByteArray",
                  "value" : "4f757470757420616c7265616479207370656e74"
               }
            ],
            "notifications" : []
         }
      ]
   },
   "id" : 1,
   "jsonrpc" : "2.0"
}
```

**Comment by fyrchik** [comment]
It is kind of strange, because I have tested it. If you have done it from scratch, is it the first storage mismatch?

**Comment by roman-khimov** [comment]
Yep, that was a restore from scratch and it's the first one to differ.

**Comment by fyrchik** [comment]
Oh, seems like it was introduced during fixing review comments.

**Comment by roman-khimov** [comment]
Confirming #984 fix, I'm at 4.3M now with no errors.
